### DISCUSSION TITLE
hello, I'm implementing a plugin that allows you to open the door that the player is looking at.

### DISCUSSION THREAD

**Discussion by barashF** [SmartlyDressedGames/U3-Docs#260] [Q&A] [answered]
using updateToggle(open) the door opens, but the texture stays the same. I suppose that you need to use updateState, but I can’t understand how, please tell me the implementation, what do I need to pass in the updateState arguments and how to get them.
Below is a piece of code in which I perform the opening:
if (raycast.GetComponent<InteractableDoorHinge>() != null)
                {
                    InteractableDoorHinge doorHinge = raycast.GetComponent<InteractableDoorHinge>();
                    DoorPlugin.Instance.Configuration.Instance.SaveData(raycast.parent.parent, command, caller);
                    string Permissions = "";
                    foreach (var item in command)
                    {
                        Permissions += item + ", ";
                    }

                    InteractableDoor door = doorHinge.door;
                    bool open = !door.isOpen;
                    door.updateState();
                    door.updateToggle(open);
                    

                    
                    UnturnedChat.Say(caller, ":" + doorHinge.GetInstanceID());
                }

**Comment by SDGNelson** [comment, marked as answer]
Currently the intended method is: BarricadeManager.ServerSetDoorOpen

**Comment by carlostipton1319** [comment]
To update both the door state and its texture when [best garage](https://digitechit.co.uk/how-do-you-select-the-best-garage-repair-companies/ ) using updateToggle(open), you can use the updateState function. Here's an example implementation:
